airbyte-server/src/main/java/io/airbyte/server/handlers/MigrationHandler.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.server.handlers;
+
+import io.airbyte.commons.io.ArchiveHelper;
+import io.airbyte.commons.lang.Exceptions;
+import io.airbyte.config.ConfigSchema;
+import io.airbyte.config.DestinationConnection;
+import io.airbyte.config.SourceConnection;
+import io.airbyte.config.StandardDestinationDefinition;
+import io.airbyte.config.StandardSourceDefinition;
+import io.airbyte.config.StandardSync;
+import io.airbyte.config.StandardSyncSchedule;
+import io.airbyte.config.StandardWorkspace;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.config.persistence.PersistenceConstants;
+import io.airbyte.server.converters.ConfigConverter;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MigrationHandler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MigrationHandler.class);
+  private static final String ARCHIVE_FILE_NAME = "airbyte_config_data";
+  private static final String VERSION_FILE_NAME = "VERSION";
+
+  private final String version;
+  private final ConfigRepository configRepository;
+
+  public MigrationHandler(final String version, final ConfigRepository configRepository) {
+    this.version = version;
+    this.configRepository = configRepository;
+  }
+
+  public Path exportData() {
+    try {
+      final Path tempFolder = Files.createTempDirectory("airbyte_archive");
+      final Path archive = Files.createTempFile(ARCHIVE_FILE_NAME, ".tar.gz");;
+      archive.toFile().deleteOnExit();
```

Comment:
I'm not sure this does exactly what you're hoping it will. I believe `deleteOnExit` only triggers when the JVM shuts down (i.e. we turn off the server). So all of these files will build up until the server is turned off, so we can still end up in a bad state where if you download the archive multiple times it causes problems. I don't actually know how to solve this. I'll poke around a bit and see if I can figure anything out.